airbyte-webapp/src/components/Modal/Modal.tsx

```diff
@@ -1,6 +1,6 @@
+import { Dialog } from "@headlessui/react";
```

Comment:
Dialog from Headless UI is accessible-by-default (captures focus, is tabbable, etc.).  It was simpler to implement this than to iterate on what we'd custom-built.

Note: `clear` is only used by the onboarding video.  really that feature behaves differently enough that it should be its own component.